site/source/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.rst

```diff
@@ -186,6 +186,46 @@ in :ref:`preamble-js`.
    doesn't mean a lot to me.
 
 
+Interacting with an API written in C/C++ from NodeJS
+====================================================
+
+Say you have a C library that exposes some procedures:
+
+.. code:: c
+
+    //api_example.c
+    #include<stdio.h>
+
+    __attribute__((used))
+    char* sayHi() {
+      return "Hi!";
+    }
+
+    __attribute__((used))
+    int daysInWeek() {
+      return 7;
+    }
+
+Compile the library with emcc:
+
+.. code:: bash
+
+    emcc api_example.c
+
+Require the library and call it's procedures from node:
+
+.. code:: javascript
+
+    var em_module = require('./a.out.js');
```

Comment:
yes, this is needed and shouldn't be a problem on windows, node will normalise this.